packages/client/src/transformer/transformer.test.tsx

```diff
@@ -520,5 +520,69 @@ describe('when draft data is transformed to graphql', () => {
         primaryCaregiverData
       )
     })
+
+    it('transform gql data from form correction data', () => {
+      const data = {
+        child: childDetails,
+        father: fatherDetails,
+        mother: motherDetails,
+        registration: {
+          ...registrationDetails,
+          contactPoint: {
+            value: 'OTHER',
+            nestedFields: {
+              contactRelationshipOther: 'grandma',
+              registrationPhone: '01736478896'
+            }
+          }
+        },
+        primaryCaregiver,
+        documents: { imageUploader: '' }
+      }
+
+      const originalData = {
+        child: {
+          ...childDetails,
+          familyNameEng: 'Khan'
+        },
+        father: fatherDetails,
+        mother: motherDetails,
+        registration: registrationDetails,
+        primaryCaregiver,
+        documents: { imageUploader: '' }
+      }
+
+      const transformedCorrectionData = {
+        values: [
+          {
+            fieldName: 'contactPoint.nestedFields.contactRelationshipOther',
+            newValue: 'grandma',
+            oldValue: 'grandma',
```

Comment:
newValue & oldValue are same?

Reply:
Thanks, @Zangetsu101 great spot! The transformer pushes the changes to the fields where `hasFieldChanged` returns true. I figured in the case of nested fields when one nested field is changed, `hasFieldChanged` returns true for another field of the same parent also. I just copied my actual transformed payload and pasted it here in the tests.

Reply:
So we can't check for `hasNestedFieldChanged` inside `hasFieldChanged` if we want to avoid this?

Reply:
Yes I think so. I think the function serves its purpose right... maybe we need an additional check in case of iterating nested field? What do you think?

Reply:
Yeah I think so too like if a field has nestedFields then we can check each nestedField for any changes and for the parent field, we only check its value and not its nestedFields?

Reply:
@Zangetsu101 Could you please take a look at [6221bbb](https://github.com/opencrvs/opencrvs-core/pull/2585/commits/6221bbb8778dfaa66df3e601390e48f05dcf92e7)?

Reply:
Looks good to me.